randovania/games/deltarune/patcherfolder/__init__.py

```diff
@@ -0,0 +1,216 @@
+        """
+        Deletes any copy of the game in the internal storage.
+        """
+        pass
+
+    def default_output_file(self, seed_hash: str) -> str:
+        """
+        Provides a output file name with the given seed hash.
+        :param seed_hash:
+        :return:
+        """
+        return f"Deltarune Randomizer"
+        
+    @property
+    def requires_input_file(self) -> bool:
+        return True
+
+    @property
+    def valid_input_file_types(self) -> List[str]:
+        return [""]
+
+    @property
+    def valid_output_file_types(self) -> List[str]:
+        return [""]
+
+    def create_patch_data(self, description: LayoutDescription, players_config: PlayersConfiguration,
+                          cosmetic_patches: DeltaruneCosmeticPatches) -> dict:
+        """
+        Creates a JSON serializable dict that can be used to patch the game.
+        Intended to be ran on the server for multiworld.
+        :return:
+        """
+        patches = description.all_patches[players_config.player_index]
+        db = default_database.game_description_for(RandovaniaGame.DELTARUNE)
+        preset = description.get_preset(players_config.player_index)
+        configuration = typing.cast(DeltaruneConfiguration, preset.configuration)
+        rng = Random(description.get_seed_for_player(players_config.player_index))
+        
+        if players_config.is_multiworld:
+            raise NotImplementedError("Multiworld is not supported for DELTARUNE")
+        useless_target = PickupTarget(pickup_creator.create_nothing_pickup(db.resource_database),
+                                      players_config.player_index)
+
+        pickup_list = pickup_exporter.export_all_indices(
+            patches,
+            useless_target,
+            db.world_list,
+            rng,
+            configuration.pickup_model_style,
+            configuration.pickup_model_data_source,
+            exporter=pickup_exporter.create_pickup_exporter(db, pickup_exporter.GenericAcquiredMemo(), players_config),
+            visual_etm=pickup_creator.create_visual_etm(),
+        )
+
+        starting_point = patches.starting_location
+
+        starting_area = db.world_list.area_by_area_location(starting_point)
+
+        starting_location_info = {
+            "starting_region": starting_point.world_name,
+        }
+        return {
+            "pickups": [
+                deltarune_pickup_details_to_patcher(detail)
+                for detail in pickup_list
+            ],
+            "starting_items": [
+                deltarune_starting_items_to_patcher(item, qty)
+                for item, qty in patches.starting_items.items()
+            ],
+            "starting_conditions": starting_location_info,
+            "has_spoiler": description.has_spoiler,
+            "description": "{}".format(description.shareable_word_hash)
+        }
+    def patch_game(self, input_file: Optional[Path], output_file: Path, patch_data: dict,
```

Comment:
the majority of this function, as well as the `deltapatcher` folder you've added, should probably be in a separate repo and uploaded as a PyPi package for use with pip. if you need assistance getting this set up let us know in discord